File: dart_sample/lib/counter_app_model.dart
import 'package:flutter/foundation.dart';

class CounterAppModel extends ChangeNotifier {
  int _count = 0;
  int _nextId = 5;
  bool _placeGlobalOnLeft = true;
  List<int> _items = <int>[1, 2, 3, 4];

  int get count => _count;

  List<int> get items => List<int>.unmodifiable(_items);

  bool get placeGlobalOnLeft => _placeGlobalOnLeft;

  Object get globalBadgeIdentity => _globalBadgeIdentity;
  final Object _globalBadgeIdentity = Object();

  void increment() {
    _count += 1;
    notifyListeners();
  }

  void decrement() {
    _count -= 1;
    notifyListeners();
  }

  void reverseItems() {
    _items = _items.reversed.toList(growable: true);
    notifyListeners();
  }

  void insertHead() {
    _items.insert(0, _nextId++);
    notifyListeners();
  }

  void removeTail() {
    if (_items.isEmpty) {
      return;
    }

    _items.removeLast();
    notifyListeners();
  }

  void toggleGlobalPlacement() {
    _placeGlobalOnLeft = !_placeGlobalOnLeft;
    notifyListeners();
  }
}

File: src/Sample/Flutter.Net/CounterAppModel.cs
using System.Collections.Generic;
using System.Linq;
using Flutter.Foundation;

// Dart parity source (reference): dart_sample/lib/counter_app_model.dart (exact sample parity)

namespace Flutter.Net;

public sealed class CounterAppModel : ChangeNotifier
{
    private int _count;
    private int _nextId = 5;
    private bool _placeGlobalOnLeft = true;
    private List<int> _items = [1, 2, 3, 4];

    public int Count => _count;

    public IReadOnlyList<int> Items => _items;

    public bool PlaceGlobalOnLeft => _placeGlobalOnLeft;

    public object GlobalBadgeIdentity { get; } = new();

    public void Increment()
    {
        _count += 1;
        NotifyListeners();
    }

    public void Decrement()
    {
        _count -= 1;
        NotifyListeners();
    }

    public void ReverseItems()
    {
        _items = [.._items.AsEnumerable().Reverse()];
        NotifyListeners();
    }

    public void InsertHead()
    {
        _items.Insert(0, _nextId++);
        NotifyListeners();
    }

    public void RemoveTail()
    {
        if (_items.Count == 0)
        {
            return;
        }

        _items.RemoveAt(_items.Count - 1);
        NotifyListeners();
    }

    public void ToggleGlobalPlacement()
    {
        _placeGlobalOnLeft = !_placeGlobalOnLeft;
        NotifyListeners();
    }
}
